In [9]:
import pandas as pd

# Read the CSV file, skipping lines with parsing errors
data = pd.read_excel('/content/Product_Rating_last_3_month_data.xlsx')

# Display the first few rows of the DataFrame
data.head()


id Product_id Product_flavor_id           Shopper_id  User_id  \
0  78602       7126             14960  1835820409595680000  1000268   
1  78603       7521             15874  4041237846491919872  1000268   
2  78604       6869             14289  2070306052036339968  1000268   
3  78605       6364             14296  3509748335361169920  1000563   
4  78606       7456             15715        1586268211147  1000563   

  Orderitem_id                                        description  \
0      1095787                                                NaN   
1      1092834                              Excellent thanks ����   
2       893829                                    Fully satiafied   
3      1110946  Thank you priceoye for providing me with A24 f...   
4      1099255  This is my 4th purchase from PriceOye._x000D_\...   

                                              images rating    rating_reason  \
0                                                 []      5    Product Price   
1                                                 []      5    Product Price   
2                                                 []      5  Product Quality   
3                                                 []      4    Product Price   
4   You should give estimate of 7 days in the fir...    NaN              NaN   

   ... Unnamed: 27 Unnamed: 28 Unnamed: 29 Unnamed: 30 Unnamed: 31  \
0  ...         NaN         NaN         NaN         NaN         NaN   
1  ...         NaN         NaN         NaN         NaN         NaN   
2  ...         NaN         NaN         NaN         NaN         NaN   
3  ...         NaN         NaN         NaN         NaN         NaN   
4  ...         NaN         NaN         NaN         NaN         NaN   

  Unnamed: 32 Unnamed: 33 Unnamed: 34 Unnamed: 35 Unnamed: 36  
0         NaN         NaT         NaN         NaT         NaT  
1         NaN         NaT         NaN         NaT         NaT  
2         NaN         NaT         NaN         NaT         NaT  
3         NaN         NaT         NaN         NaT         NaT  
4         NaN         NaT         NaN         NaT         NaT  

[5 rows x 37 columns]

In [10]:


# Assuming your DataFrame is named 'data' and the column containing Product IDs is 'Product_id'
top_product_counts = data['Product_id'].value_counts().head(10)

# Display the top 3 Product ID counts
print(top_product_counts)


Product_id
7126    429
6194    296
7040    258
7457    190
6747    169
6163    146
5936    135
6640    132
6827    123
7865    107
Name: count, dtype: int64


In [18]:
import pandas as pd

# Read the CSV file, skipping lines with parsing errors
import pandas as pd

# Read the Excel file into a DataFrame
data = pd.read_excel('/content/Jan to March Ratings Data.xlsx')
data.head()


id Product_id Product_flavor_id           Shopper_id  User_id  \
0  69739       6796             15159  6338974497501230080  1000268   
1  69740       7506             15830  1494924265562599936  1000268   
2  69741       6214             12894  2411044390941850112  1000268   
3  69742       4504             10681        1646770513608  1000268   
4  69743       7055             14879  4126718437784010240  1000563   

  Orderitem_id                               description images rating  \
0       970105                                       NaN     []      4   
1       979718                                       NaN     []      5   
2       922650              In this price a good product     []      5   
3       387710  Nice product & service is also good ����     []      5   
4       945363                                       NaN     []      2   

      rating_reason  ... Unnamed: 27 Unnamed: 28 Unnamed: 29 Unnamed: 30  \
0  Delayed Delivery  ...         NaN         NaN         NaN         NaN   
1   Product Quality  ...         NaN         NaN         NaN         NaN   
2     Product Price  ...         NaN         NaN         NaN         NaN   
3   Product Quality  ...         NaN         NaN         NaN         NaN   
4         Packaging  ...         NaN         NaN         NaN         NaN   

  Unnamed: 31 Unnamed: 32 Unnamed: 33 Unnamed: 34 Unnamed: 35 Unnamed: 36  
0         NaN         NaN         NaT         NaN         NaT         NaT  
1         NaN         NaN         NaT         NaN         NaT         NaT  
2         NaN         NaN         NaT         NaN         NaT         NaT  
3         NaN         NaN         NaT         NaN         NaT         NaT  
4         NaN         NaN         NaT         NaN         NaT         NaT  

[5 rows x 37 columns]

**List down Top Products getting Reviews, having Rating 5-Star**

In [19]:
data['rating'].fillna('', inplace=True)

# Filter the DataFrame to include only rows where the 'rating' column contains '5'
five_star = data[data['rating'] == 5]


# Display the filtered DataFrame
#print(five_star)

top_product_counts = five_star['Product_id'].value_counts().head(3)

# Display the top 3 Product ID counts
print(top_product_counts)


Product_id
7126    398
6194    324
6640    295
Name: count, dtype: int64


Top Products getting Reviews, having Rating 5-Star
April 1 to 22
Xiomai Redmi 12 (Reviews: 93)
Infinix Note 30 Pro (Reviews: 62)
Tecno Spark 10 Pro (Reviews: 61)
1st Jan to 31st March
Xiomai Redmi 12 (Reviews: 398)
M10 TWS Wireless Bluetooth Earbuds (Reviews: 324)
Xiaomi Redmi Note 12 (Reviews: 295)

Now, On the basis of above Product_ids, I'll able to identify Products from Product table (Product table and Product Rating having 1 to Many Relationship in database) in my database.
List of those are as follows:
1. Xiaomi Redmi 12
2. M10 TWS Wireless Bluetooth Earbuds
3. Air 31 TWS Transparent Earbuds

Afterwards, getting this inside We maintain more stocks for above products and Do Marketing in a more appropriate way to grab clients.